### PR TITLE
vcenter input/processor and grafana output for vc input

### PR DESCRIPTION
**Pull request by askripe**
grafana output for vc input updated
add vc input
add vcenter processore
remove vcenter package move filtering logic into template
errors fixing for telegram output
some error handling
rewrite vcenter processor
add sample events from vcenter
add rate limiter
add debug print for spaning errors
add vcenter events output into grafana templates
finalize telegram templates
grafana will add annotation instead of region annotation
update go.mod/go.sum

typos and warnings

implement some events

### CODE REVIEW

File: output/grafana.go [hunk 1]
```diff
@@ -118,7 +124,7 @@ func (g *GrafanaOutput) Send(event *common.Event) {
 			g.logger.SpanError(span, err)
 		}
 
-		err = g.grafanaEventer.At(message, attributes, event.Time)
+		err = g.grafanaEventer.Interval(message, attributes, event.Time, event.Time)
```
Comment (XCiber) [inline output/grafana.go:127]: return At for now pls

**Review by XCiber — CHANGES_REQUESTED** [review]

File: processor/vcenter.go [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+		"AlarmSnmpCompletedEvent":
+		vce.DestLocation, _ = jsonparser.GetString(jsonByte, "data", "ComputeResource", "Name")
+		vce.DestClusterName, _ = jsonparser.GetString(jsonByte, "data", "Datacenter", "Name")
+		vce.DestDatacenterName, _ = jsonparser.GetString(jsonByte, "data", "Datacenter", "Datacenter", "Value")
+		vce.DestESXiHostName, _ = jsonparser.GetString(jsonByte, "data", "Host", "Name")
+		vce.EntityName, _ = jsonparser.GetString(jsonByte, "data", "Source", "Entity", "Value")
+		vce.EntityType, _ = jsonparser.GetString(jsonByte, "data", "Source", "Entity", "Type")
+		vce.AlarmName, _ = jsonparser.GetString(jsonByte, "data", "Alarm", "Name")
+		return nil
+		//
```
[the remark this answers is not in the document]
Comment (askripe) [reply processor/vcenter.go]: fixed

File: grafana.attributes [hunk 3]
```diff
@@ -9,4 +9,20 @@
   {{- end}}
 
 {{- end}}
-{{- define "grafana-attributes"}}{{- if .data}}{{template "text" .}}{{end}}{{- end}}
\ No newline at end of file
+{{- define "vcenter"}}
```
[the remark this answers is not in the document]
Comment (askripe) [reply grafana.attributes]: fixed

File: grafana.message [hunk 4]
```diff
@@ -10,15 +10,23 @@
     {{- end}}
   {{- end}}
 {{- end}}
-
+{{- define "vcenter"}}
```
[the remark this answers is not in the document]
Comment (askripe) [reply grafana.message]: fixed

**Review by XCiber — APPROVED** [review]

**Pull request merged**